tests/workflows/test_ao_ic_usd.py

```diff
@@ -0,0 +1,58 @@
+import os
+from argparse import Namespace
+
+import pandas as pd
+
+from mawpy.constants import (USER_ID, UNIX_START_T, UNIX_START_DATE, ORIG_LAT, ORIG_LONG, ORIG_UNC, STAY_LAT, STAY_LONG,
+                             STAY_UNC, STAY, STAY_DUR)
+from mawpy.steps.address_oscillation import address_oscillation
+from mawpy.steps.incremental_clustering import incremental_clustering
+from mawpy.steps.update_stay_duration import update_stay_duration
+from mawpy.workflows.ao_ic_usd import ao_ic_usd
+
+
+def test_workflow_ao_ic_usd(tmp_path):
+
+    input_file = os.path.dirname(__file__) + '/../resources/test_input.csv'
+    output_file = os.path.join(tmp_path, 'test_output_workflow_ao_ic_usd.csv')
+    # Prepare arguments similar to command-line arguments
+    args = Namespace(
+        input_file=input_file,
+        output_file=output_file,
+        spatial_constraint=1,
+        duration_constraint_1=0,
+        duration_constraint_2=300,
+        duration_constraint_3=300
+    )
+
+    # Run the workflow to be tested and get the output
+    actual_output_df = ao_ic_usd(args.input_file, args.output_file, args.spatial_constraint,
+                                 args.duration_constraint_1, args.duration_constraint_2, args.duration_constraint_3)
+
+    # Check if output file was created
+    assert os.path.exists(output_file)
+
+    # Destroy test output file after test completed
+    os.remove(args.output_file)
+
+    # Check if the workflow steps were called with expected arguments
+    df_output_ao = address_oscillation(output_file, args.duration_constraint_1,
+                                       input_file=args.input_file)
+    df_output_ic = incremental_clustering(output_file, args.spatial_constraint, args.duration_constraint_2,
+                                          input_df=df_output_ao)
+    expected_output_df = update_stay_duration(output_file, args.duration_constraint_3, input_df=df_output_ic)
+
+    # Assert if the expected_output_df equal to actual_output_df
+    pd.testing.assert_frame_equal(actual_output_df, expected_output_df)
```

Comment:
are we simply testing that `ao_ic_usd` calls the 3 steps? There's not much value there. Could you have a verified and saved `test_output.csv` and compare to that instead?